File: backend/app/schemas/claim.py
"""Claim Pydantic schemas."""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID


class ClaimBase(BaseModel):
    """Base claim schema."""
    claim_text: str
    claim_type: Optional[str] = None
    context: Optional[str] = None
    is_checkable: bool = True


class ClaimCreate(ClaimBase):
    """Schema for creating a claim."""
    article_id: UUID
    extraction_confidence: Optional[float] = None


class ClaimResponse(ClaimBase):
    """Schema for claim response."""
    id: UUID
    article_id: UUID
    extraction_confidence: Optional[float]
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class ClaimDetailResponse(ClaimResponse):
    """Schema for detailed claim response."""
    article_title: Optional[str] = None
    investigation: Optional[dict] = None

File: frontend/src/types/claim.ts
export interface Claim {
  id: string;
  claim_text: string;
  claim_type?: string;
  context?: string;
  is_checkable: boolean;
  extraction_confidence?: number;
  status: string;
  investigation?: any;
  created_at: string;
}

export type Verdict = 'true' | 'mostly_true' | 'mixed' | 'mostly_false' | 'false' | 'unverifiable';
